modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,20 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev Safe account has two types of signatures: EOA and Smart Contract signatures. While the EOA signature is 
+     * fixed in size, the Smart Contract signature can be of arbitrary length. Safe encodes the Smart Contract
+     * signature length in the signature data. A malicious bundler can pad additional bytes to the `signature` data,
```

Comment:
I still find this a bit confusing :see_no_evil: - in particular, we call the group of owner signatures for the Safe `signatures`, but here `signature` (I _think_. At first, I thought this was referring to the data for a single Smart Contract signature, but then realized it wasn't).

In particular, this sentence still reads like it is possible to pad `signatures` bytes, but it isn't.

Reply:
https://github.com/safe-global/safe-modules/pull/471/commits/a50f197ad3c46ecb36e61b15466ce958bab26167